pkg/virt-handler/non-root.go

```diff
@@ -135,3 +136,49 @@ func (d *VirtualMachineController) nonRootSetup(origVMI, vmi *v1.VirtualMachineI
 	}
 	return nil
 }
+
+func (d *VirtualMachineController) ensureDVsOwnership(vmi *v1.VirtualMachineInstance) error {
+	res, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+	passthoughFSVolumes := make(map[string]struct{})
+	for i := range vmi.Spec.Domain.Devices.Filesystems {
+		passthoughFSVolumes[vmi.Spec.Domain.Devices.Filesystems[i].Name] = struct{}{}
+	}
+
+	pvcVolume := make(map[string]v1.VolumeStatus)
+	hotplugVolumes := make(map[string]bool)
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.HotplugVolume != nil {
+			hotplugVolumes[volumeStatus.Name] = true
+		}
+
+		if volumeStatus.PersistentVolumeClaimInfo != nil {
+			pvcVolume[volumeStatus.Name] = volumeStatus
+		}
+	}
+
+	for i := range vmi.Spec.Volumes {
+		volume := vmi.Spec.Volumes[i]
+		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.DataVolume != nil {
+			if _, isPassthoughFSVolume := passthoughFSVolumes[volume.Name]; isPassthoughFSVolume {
+				log.Log.V(4).Infof("this volume %s is mapped as a filesystem passthrough, skipping", volume.Name)
+				continue
+			}
+
+			if hotplugVolumes[volume.Name] {
+				log.Log.V(4).Infof("this volume %s is hotplugged, skipping", volume.Name)
+				continue
```

Comment:
why skip hotplug? don't hotplug disks potentially have the same file permissions issue?

Reply:
It's a different code path and therefore it needs more tweaks. I already have a test for this but I want to PR it separately if you don't mind.